packages/grpc-js/src/server.ts

```diff
@@ -500,10 +500,12 @@ export class Server {
       }
     }
 
-    // If any sessions are active, close them gracefully.
     pendingChecks += this.sessions.size;
     this.sessions.forEach((session) => {
-      session.close(maybeCallback);
+      if (!session.closed) {
+        pendingChecks += 1;
```

Comment:
Took me a second to realize why this was added. I don't like the fact there's a disjoint between adding a new pending action, and incrementing this counter: this feels brittle. But I don't see a way to do this without over engineering it completely:

```javascript
  function addAction(action) {
    pendingChecks++;
    action(maybeCallback);
}

[...]

addAction(callback => session.close(callback));
```

Reply:
I don't know if this is worth changing here but I'm glad you brought it up because there was actually an error. See the new commit.